### PR TITLE
[MOC-62][MOC-59] Recording state fixes for sign up and getting recordings for first page

### PR DESCRIPTION
**Pull request by nicolaschapur**
This PR adds a bunch o fixes for recording states and signed or not users. Also adds a recording call when opening the extension, to check if we need to show the list of recordings as the first page or the record button.

https://github.com/Mocksi/HARlighter/assets/95246792/762dd263-3cf2-4061-87e4-406d5e864139

## Summary by CodeRabbit

- **New Features**
  - Improved retrieval and management of recordings with updated logic.

- **Bug Fixes**
  - Enhanced handling of recordings retrieval to ensure correct storage and display.

- **Refactor**
  - Streamlined recording state management based on user login status.
  - Updated styling class names for better readability and maintenance.

- **Style**
  - Adjusted CSS class names for consistency and improved styling.

- **Performance**
  - Optimized recordings fetching by removing unnecessary data parsing.

### CODE REVIEW

File: apps/mocksi-lite/content/content.tsx [hunk 1]
```diff
@@ -39,50 +40,71 @@ function initial() {
...
+			window.location.origin !== SignupURL
+		) {
+			window.open(SignupURL);
+		}
+		sendMessage("resetIcon");
+		state = RecordingState.UNAUTHORIZED;
+	}
+
+	return { state, email };
+}
```
[the remark this answers is not in the document]
Comment (nicolaschapur) [reply apps/mocksi-lite/content/content.tsx:93]: in the case that recordings is undefined or null the getRecordingsStorage function itself return and empty array

File: apps/mocksi-lite/background.ts [hunk 2]
```diff
@@ -97,9 +103,27 @@ chrome.action.onClicked.addListener((activeTab) => {
 			onAttach.bind(null, currentTabId),
 		);
 		chrome.debugger.onDetach.addListener(debuggerDetachHandler);
-		chrome.tabs.sendMessage(currentTabId || 0, {
-			text: "clickedIcon",
-		});
+		getRecordingsStorage()
+			.then((recordings) => {
```
Comment (elg0nz) [inline apps/mocksi-lite/background.ts:107]: let's refactor this.
I cannot see any reason to call 
```tsx
chrome.tabs.sendMessage(currentTabId || 0, {
 						text: "clickedIcon",
 					});
```
three times, with the same parameters.
Specially when you can use [finally](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) in this promise

File: apps/mocksi-lite/content/Popup/CreateDemo/index.tsx [hunk 3]
```diff
@@ -28,7 +28,11 @@ const CreateDemo = ({
 			try {
 				const newRecordings = await getRecordingsStorage();
 				if (newRecordings.length !== recordings.length) {
-					setRecordings(newRecordings);
+					const sortedRecordings = newRecordings.sort(
+						(a: Recording, b: Recording) =>
+							a.updated_timestamp > b.updated_timestamp ? -1 : 0,
+					);
+					setRecordings(sortedRecordings);
```
[the remark this answers is not in the document]
Comment (elg0nz) [reply apps/mocksi-lite/content/Popup/CreateDemo/index.tsx:35]: This is a good suggestion

File: apps/mocksi-lite/content/content.tsx [hunk 4]
```diff
@@ -39,50 +40,71 @@ function initial() {
...
+}> {
+	const email = await getEmail();
+	const results = await chrome.storage.local.get([MOCKSI_RECORDING_STATE]);
+	const recordings = await getRecordingsStorage();
+
+	const recordingState = results[MOCKSI_RECORDING_STATE];
+	let state = recordingState;
+	console.log({ recordingState });
+
+	if (email) {
```
Comment (elg0nz) [inline apps/mocksi-lite/content/content.tsx:55]: this function is already too long. 
Extract this out so you can do:
`tsx
state = email ? setLoggedInsState() : setLoggedOutState()

return { state, email };
}
```

Where `setLoggedOutState` would look like this:
```tsx
		if (
 			recordingState === RecordingState.UNAUTHORIZED &&
 			window.location.origin !== SignupURL
 		) {
 			window.open(SignupURL);
 		}
 		sendMessage("resetIcon");
 		return RecordingState.UNAUTHORIZED;
```

**Review by elg0nz — CHANGES_REQUESTED** [review]
Still needs some work

**Pull request closed without merging**